perfkitbenchmarker/flag_util.py

```diff
@@ -219,3 +220,49 @@ def __enter__(self):
   def __exit__(self, *unused_args, **unused_kwargs):
     """Stops the flag substitution."""
     self._flags.__dict__['FlagDict'] = self._original_flagdict
+
+
+class UnitsParser(flags.ArgumentParser):
+  """Parse a flag containing a unit expression."""
+
+  syntactic_help = 'A quantity with a unit. Ex: 12MB'
+
+  def Parse(self, inp):
+    """Parse the input.
+
+    Args:
+      inp: a string
+
+    Returns:
+      A pint.Quantity.
+
+    Raises:
+      ValueError if it can't parse its input.
+    """
+
+    if isinstance(inp, pint_wrapper.ureg.Quantity):
+      return inp
+
+    try:
+      return pint_wrapper.ureg.parse_expression(inp)
+    except Exception as e:
```

Comment:
I know catching all exceptions is ugly. My concern here is that I don't know what exceptions `parse_expression` might throw (I'm pretty sure it can be more than `UndefinedUnitError`), but I want all errors in the flag parser to raise a `ValueError`. Is there a nicer way to do that?